### PR TITLE
listener for list of questions and course

### LINKED ISSUE #6: [Utilities] Set up Event Listeners for Courses and Questions
## Description
As we want live updates for the queue, we want to listen 👂 to any changes of a course's Questions. 
Live updates to general Course information also makes sense - onDuty, tags, location namely.
image
Create hook(s) to manage live data onto the frontend! 

https://firebase.google.com/docs/firestore/query-data/listen

### PR DESCRIPTION
**Pull request by lnguyen2693**

### CODE REVIEW

File: src/app/hooks/oh/useCourseLoader.tsx [hunk 1]
```diff
@@ -22,22 +26,29 @@ export const useCourseLoader = (props: UseCourseLoaderProps) => {
...
+      (snapshot) => {
+        if (snapshot.exists()) {
+          setValue({ id: snapshot.id, ...snapshot.data() });
+        }
+      }
+    );
+
+    unsubscriber.current = unsubscribe;
+
+    return () => unsubscribe();
```
Comment (nickbar01234) [inline src/app/hooks/oh/useCourseLoader.tsx]: Should it be `unsubscriber.current()`?

File: src/app/hooks/oh/useQuestionsLoader.tsx [hunk 2]
```diff
@@ -11,21 +16,46 @@ interface UseQuestionsLoaderProps {
...
+          questionsList.push({ id: doc.id, ...doc.data() });
+        });
+
+        setValue(questionsList);
+      }
+    );
+
+    unsubscriber.current = unsubscribe;
+
+    return () => unsubscribe();
```
Comment (nickbar01234) [inline src/app/hooks/oh/useQuestionsLoader.tsx]: Same thing here

File: src/app/hooks/oh/useCourseLoader.tsx [hunk 3]
```diff
@@ -22,22 +26,29 @@ export const useCourseLoader = (props: UseCourseLoaderProps) => {
...
-  //     students: [],
-  //     tas: [],
-  //     onDuty: [],
-  //     tags: {},
-  //   });
-  // }, []);
+
+    // listener for course
+    const unsubscribe = onSnapshot(
+      doc(db, `courses/${props.courseId}`).withConverter(courseConverter),
```
Comment (johnny-t06) [inline src/app/hooks/oh/useCourseLoader.tsx]: small nit: can we destructure the props

**Review by johnny-t06 — APPROVED** [review]
LGTM, logic makes sense! Feel free to merge with the small comments

**Pull request merged**